org.sailfishos.usermanager.xml

```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE node PUBLIC "-//freedesktop//DTD D-BUS Object Introspection 1.0//EN"
+"http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd">
+<node>
+  <interface name="org.sailfishos.usermanager">
+    <method name="users">
+        <arg direction="out" type="a(ssi)" name="users"/>
+        <annotation name="org.qtproject.QtDBus.QtTypeName.Out0" value="QList&lt;SailfishUserManagerEntry&gt;"/>
+    </method>
+    <method name="addUser">
+        <arg direction="in" type="s" name="user_name"/>
+        <arg direction="in" type="s" name="full_name"/>
+    </method>
+    <signal name="userAdded">
+        <arg type="(ssi)" name="user"/>
+        <annotation name="org.qtproject.QtDBus.QtTypeName.Out0" value="SailfishUserManagerEntry"/>
+    </signal>
+    <method name="removeUser">
+        <arg direction="in" type="s" name="user_name"/>
```

Comment:
If uid is the (immutable) key in the user database, then the caller must pass uid here rather than username (which can change between the time when you looked it up and the time when this call reaches the service)